pkg/controller/acmechallenges/sync_test.go

```diff
@@ -85,8 +86,49 @@ func TestSyncHappyPath(t *testing.T) {
 		}),
 		gen.SetChallengeFinalizers([]string{cmacme.ACMEFinalizer}),
 	)
-
+	deletedChallenge := gen.ChallengeFrom(baseChallenge,
+		gen.SetChallengeDeletionTimestamp(metav1.Now()))
 	tests := map[string]testT{
+		"cleanup if the challenge is deleted and remove the finalizer": {
```

Comment:
Suggestion: I can think of two more cases that might be useful to test
- if a challenge has a finalizer, but it's not `cmacme.ACMEFinalizer` (and there is no deletion timestamp), `cmacme.ACMEFinalizer` gets added
- if there is a delete timestamp and an `cmacme.ACMEFinalizer` finalizer as well as some other finalizer, only the `cmacme.ACMEFinalizer` gets removed

(I am thinking that there might be use cases where users add custom finalizers if they want to perform some custom actions to cleanup or do somethig else before a challenge gets deleted- not sure if that actually ever is the case)

Reply:
Ahh, I just read the PR description properly.

> Challenge Finalizer is assumed to be the only (first) finalizer (breaks if external controllers add their own finalizers)

This is then maybe a regression in v1.8.0 as I think this would have been working in v1.7.0 where the finalizer was being added by the orders controller at the challenge creation time.
I have not tested what the behaviour was in v1.7 vs v1.8, but I guess if it was possible to add another finalizer then and is not possible now, we should fix it and cut a patch